File: program/java/lfz oo/sourecode/ch4/SwapDemo.cpp
// Swap coordinates -- C++ °æ±¾
#include <iostream>
using namespace std;

class Coord {
	public:
		int x;
		int y;
};

// Swap contents of two Coord objects.
void swap(Coord &a, Coord &b) {
	Coord temp;
	
	// swap contents of objects
	temp = a;
	a = b;
	b = temp;
}

int main(){
	Coord ob1, ob2;
	
	ob1.x = 10;
	ob1.y = 20;
	ob2.x = 88;
	ob2.y = 99;
	
	cout << "Original values:\n";
	cout << "ob1: " << ob1.x << ", " << ob1.y << "\n";
	cout << "ob2: " << ob2.x << ", " << ob2.y << "\n";

	swap(ob1, ob2);
	
	cout << "Swapped values:\n";
	cout << "ob1: " << ob1.x << ", " << ob1.y << "\n";
	cout << "ob2: " << ob2.x << ", " << ob2.y << "\n";
	
	return 0;
}

File: program/java/lfz oo/sourecode/源代码/ch4/SwapDemo.java
//: SwapDemo.java

//package g3ds.joop.ch4;

class Coord {
	int x;
	int y;
}

public class SwapDemo{
	static void swap(Coord a, Coord b) {
		Coord temp = new Coord();

		temp = a;
		a = b;
		b = temp;
	}

	public static void main(String[] args) {
		Coord ob1 = new Coord();
		Coord ob2 = new Coord();

		ob1.x = 10;
		ob1.y = 20;

		ob2.x = 88;
		ob2.y = 99;

		System.out.println("Original values:");
		System.out.println("ob1: " + ob1.x + ", " + ob1.y);
		System.out.println("ob2: " + ob2.x + ", " + ob2.y);

		swap(ob1, ob2);

		System.out.println("Swapped values:");
		System.out.println("ob1: " + ob1.x + ", " + ob1.y);
		System.out.println("ob2: " + ob2.x + ", " + ob2.y);
	}
}